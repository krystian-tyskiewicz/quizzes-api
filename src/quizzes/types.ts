export type Quiz = {
  name: string;
  questions: Question[];
};

export type Question = {
  title: string;
  answers: Answer[];
};

export type Answer = {
  title: string;
  isCorrect?: boolean;
};
