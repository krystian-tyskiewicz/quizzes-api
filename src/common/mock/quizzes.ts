import { Quiz } from '../../quizzes/entities/quiz.entity';
import { Question } from '../../quizzes/entities/question.entity';
import { Answer } from '../../quizzes/entities/answer.entity';

export const createQuizDto = {
  name: 'Cars Quiz',
  questions: [
    {
      name: 'What does ABS stand for in car terms?',
      answers: [
        { name: 'Anti-lock Braking System', isCorrect: true },
        { name: 'Automatic Braking System', isCorrect: false },
        { name: 'Anti-Break System', isCorrect: false },
      ],
    },
    {
      name: 'Which country is home to the car manufacturer Ferrari?',
      answers: [
        { name: 'France', isCorrect: false },
        { name: 'Italy', isCorrect: true },
        { name: 'Germany', isCorrect: false },
      ],
    },
  ],
} as Quiz;

export const getCarsQuiz = () => {
  const quiz = {
    id: 1,
    name: 'Cars Quiz',
    questions: [
      {
        id: 1,
        name: 'What does ABS stand for in car terms?',
        answers: [
          { id: 1, name: 'Anti-lock Braking System', isCorrect: true },
          { id: 2, name: 'Automatic Braking System', isCorrect: false },
          { id: 3, name: 'Anti-Break System', isCorrect: false },
        ],
      },
      {
        id: 2,
        name: 'Which country is home to the car manufacturer Ferrari?',
        answers: [
          { id: 4, name: 'France', isCorrect: false },
          { id: 5, name: 'Italy', isCorrect: true },
          { id: 6, name: 'Germany', isCorrect: false },
        ],
      },
    ],
  } as Quiz;

  return prepareQuiz(quiz);
};

export const getGeographyQuiz = () => {
  const quiz = {
    id: 2,
    name: 'Geography Quiz',
    questions: [
      {
        id: 3,
        name: 'What is the capital of Japan?',
        answers: [
          { id: 7, name: 'Beijing', isCorrect: false },
          { id: 8, name: 'Seoul', isCorrect: false },
          { id: 9, name: 'Tokyo', isCorrect: true },
          { id: 10, name: 'Kyoto', isCorrect: false },
        ],
      },
      {
        id: 4,
        name: 'Which continent is the Sahara Desert located on?',
        answers: [
          { id: 11, name: 'Asia', isCorrect: false },
          { id: 12, name: 'South America', isCorrect: false },
          { id: 13, name: 'Africa', isCorrect: true },
          { id: 14, name: 'Australia', isCorrect: false },
        ],
      },
      {
        id: 5,
        name: 'Which river flows through the city of Cairo?',
        answers: [
          { id: 15, name: 'Tigris', isCorrect: false },
          { id: 16, name: 'Nile', isCorrect: true },
          { id: 17, name: 'Amazon', isCorrect: false },
          { id: 18, name: 'Yangtze', isCorrect: false },
        ],
      },
    ],
  } as Quiz;

  return prepareQuiz(quiz);
};

const prepareQuiz = (quiz: Quiz) => {
  const questions = quiz.questions.map((question: Question) => {
    const answers = question.answers.map((answer: Answer) => ({
      ...answer,
      question,
    }));
    return { ...question, quiz, answers };
  });

  return { ...quiz, questions };
};

export const quizzes: Quiz[] = [getCarsQuiz(), getGeographyQuiz()];
export const quiz = quizzes[0];

export const updateQuizDto = {
  name: 'New Cars Quiz',
  questions: [
    {
      name: 'Which of these is considered a muscle car?',
      answers: [
        { name: 'Mini Cooper', isCorrect: false },
        { name: 'Ford Mustang', isCorrect: true },
        { name: 'Toyota Prius', isCorrect: false },
      ],
    },
    {
      name: 'In car engines, what is a turbocharger used for?',
      answers: [
        { name: 'To improve fuel efficiency', isCorrect: false },
        { name: 'To increase engine power', isCorrect: true },
        { name: 'To reduce emissions', isCorrect: false },
      ],
    },
  ],
} as Quiz;

const getUpdatedQuiz = () => {
  const quiz = {
    id: 1,
    name: 'New Cars Quiz',
    questions: [
      {
        id: 20,
        name: 'Which of these is considered a muscle car?',
        answers: [
          { id: 30, name: 'Mini Cooper', isCorrect: false },
          { id: 31, name: 'Ford Mustang', isCorrect: true },
          { id: 32, name: 'Toyota Prius', isCorrect: false },
        ],
      },
      {
        id: 21,
        name: 'In car engines, what is a turbocharger used for?',
        answers: [
          { id: 33, name: 'To improve fuel efficiency', isCorrect: false },
          { id: 34, name: 'To increase engine power', isCorrect: true },
          { id: 35, name: 'To reduce emissions', isCorrect: false },
        ],
      },
    ],
  } as Quiz;

  return prepareQuiz(quiz);
};

export const updatedQuiz = getUpdatedQuiz();
