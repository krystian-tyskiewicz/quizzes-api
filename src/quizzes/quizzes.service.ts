import { Injectable } from '@nestjs/common';
import { Quiz } from '../types';

@Injectable()
export class QuizzesService {
  getQuizzes(): Quiz[] {
    const quizzes = [
      {
        name: 'HTML',
        questions: [
          {
            title: 'What type of a language is HTML?',
            answers: [
              { title: 'Network Protocol' },
              { title: 'Programming Language' },
              { title: 'Scripting Language' },
              { title: 'Markup Language', isCorrect: true },
            ],
          },
          {
            title: 'What should be the first tag in any HTML Document?',
            answers: [
              { title: '<head>' },
              { title: '<title>' },
              { title: '<html>' },
              { title: '<!doctype html>', isCorrect: true },
            ],
          },
          {
            title: 'What tag is used to display a picture in a HTML page?',
            answers: [
              { title: 'Img' },
              { title: 'Src' },
              { title: 'Picture' },
              { title: 'Image', isCorrect: true },
            ],
          },
          {
            title: 'HTML are web pages read and rendered by',
            answers: [
              { title: 'Server' },
              { title: 'Web Browser', isCorrect: true },
              { title: 'Compiler' },
              { title: 'Interpreter' },
            ],
          },
          {
            title: 'What is the correct HTML tag for inserting a line break?',
            answers: [
              { title: '<br />', isCorrect: true },
              { title: '<lb />' },
              { title: '<break />' },
              { title: '<nbsp>' },
            ],
          },
          {
            title: 'What is the correct HTML for making a hyperlink?',
            answers: [
              { title: 'url="http://mcqsets.com">MCQ Sets Quiz' },
              { title: '<a name="http://mcqsets.com">MCQ Sets Quiz</a>' },
              {
                title: '<a href="http://mcqsets.com">MCQ Sets Quiz</a>',
                isCorrect: true,
              },
              { title: '<http://mcqsets.com</a>' },
            ],
          },
          {
            title: 'HTML documents are saved in',
            answers: [
              { title: 'None of above' },
              { title: 'ASCII text', isCorrect: true },
              { title: 'Special binary format' },
              { title: 'Machine language codes' },
            ],
          },
        ],
      },
      {
        name: 'Cars',
        questions: [
          {
            title:
              'What is the best-selling British car model of all time, known for its simple design and affordability?',
            answers: [
              { title: 'MINI', isCorrect: true },
              { title: 'Morris Minor' },
              { title: 'Austin Metro' },
            ],
          },
          {
            title:
              'Which British car brand produced the iconic DB5, featured in James Bond films such as "Goldfinger"?',
            answers: [
              { title: 'Bentley' },
              { title: 'Aston Martin', isCorrect: true },
              { title: 'Lotus' },
            ],
          },
          {
            title:
              'The Mini Cooper S is known for its rally successes in the 1960s. What is the name of the racing legend associated with driving the Mini?',
            answers: [
              { title: 'Stirling Moss' },
              { title: 'Paddy Hopkirk', isCorrect: true },
              { title: 'Jackie Stewart' },
            ],
          },
          {
            title:
              'Although the Ford Escort was incredibly popular, it wasn’t the best-selling family saloon of the 1970’s. Which of the following was?',
            answers: [
              { title: 'Austin Allegro' },
              { title: 'Vauxhall Cavalier' },
              { title: 'Ford Cortina Stewart', isCorrect: true },
            ],
          },
          {
            title: 'The ornament on the bonnet of a Rolls Royce is called…',
            answers: [
              { title: 'Spirit of Ecstasy', isCorrect: true },
              { title: 'Flying Lady' },
              { title: 'Elegant Emissary' },
            ],
          },
          {
            title:
              'The Defender is a legendary off-road vehicle produced by which British automaker?',
            answers: [
              { title: 'Land Rover', isCorrect: true },
              { title: 'Range Rover' },
              { title: 'Jaguar' },
            ],
          },
          {
            title: 'Which British manufacturer makes the model ‘Elise’?',
            answers: [
              { title: 'McLaren' },
              { title: 'Lotus', isCorrect: true },
              { title: 'Aston Martin' },
            ],
          },
        ],
      },
    ];

    return quizzes;
  }
}
