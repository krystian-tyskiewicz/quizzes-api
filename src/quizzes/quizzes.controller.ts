import { Controller, Get } from '@nestjs/common';
import { QuizzesService } from './quizzes.service';
import { Quiz } from './types';

@Controller('quizzes')
export class QuizzesController {
  constructor(private readonly quizzesService: QuizzesService) {}

  @Get()
  getQuizzes(): Quiz[] {
    return this.quizzesService.getQuizzes();
  }
}
