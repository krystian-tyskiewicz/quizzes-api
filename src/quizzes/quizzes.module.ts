import { Module } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';
import { QuizzesController } from './controllers/quizzes.controller';
import { UserQuizController } from './controllers/user-quiz.controller';
import { QuizzesService } from './services/quizzes.service';
import { UserQuizService } from './services/user-quiz.service';
import { Quiz } from './entities/quiz.entity';
import { Question } from './entities/question.entity';
import { Answer } from './entities/answer.entity';
import { UserQuiz } from './entities/user-quiz.entity';
import { UserAnswer } from './entities/user-answer.entity';

@Module({
  imports: [
    TypeOrmModule.forFeature([Quiz, Question, Answer, UserQuiz, UserAnswer]),
  ],
  controllers: [QuizzesController, UserQuizController],
  providers: [QuizzesService, UserQuizService],
})
export class QuizzesModule {}
