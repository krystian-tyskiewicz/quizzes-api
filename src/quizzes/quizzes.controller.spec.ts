import { Test, TestingModule } from '@nestjs/testing';
import { QuizzesController } from './quizzes.controller';
import { QuizzesService } from './quizzes.service';

describe('QuizzesController', () => {
  let app: TestingModule;

  beforeAll(async () => {
    app = await Test.createTestingModule({
      controllers: [QuizzesController],
      providers: [QuizzesService],
    }).compile();
  });

  describe('getQuizzes', () => {
    it('should return quizzes', () => {
      const quizzesController = app.get(QuizzesController);
      expect(quizzesController.getQuizzes()).toMatchSnapshot();
    });
  });
});
