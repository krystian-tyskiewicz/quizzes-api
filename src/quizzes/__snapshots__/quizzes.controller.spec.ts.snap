// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizzesController getQuizzes should return quizzes 1`] = `
[
  {
    "name": "HTML",
    "questions": [
      {
        "answers": [
          {
            "title": "Network Protocol",
          },
          {
            "title": "Programming Language",
          },
          {
            "title": "Scripting Language",
          },
          {
            "isCorrect": true,
            "title": "Markup Language",
          },
        ],
        "title": "What type of a language is HTML?",
      },
      {
        "answers": [
          {
            "title": "<head>",
          },
          {
            "title": "<title>",
          },
          {
            "title": "<html>",
          },
          {
            "isCorrect": true,
            "title": "<!doctype html>",
          },
        ],
        "title": "What should be the first tag in any HTML Document?",
      },
      {
        "answers": [
          {
            "title": "Img",
          },
          {
            "title": "Src",
          },
          {
            "title": "Picture",
          },
          {
            "isCorrect": true,
            "title": "Image",
          },
        ],
        "title": "What tag is used to display a picture in a HTML page?",
      },
      {
        "answers": [
          {
            "title": "Server",
          },
          {
            "isCorrect": true,
            "title": "Web Browser",
          },
          {
            "title": "Compiler",
          },
          {
            "title": "Interpreter",
          },
        ],
        "title": "HTML are web pages read and rendered by",
      },
      {
        "answers": [
          {
            "isCorrect": true,
            "title": "<br />",
          },
          {
            "title": "<lb />",
          },
          {
            "title": "<break />",
          },
          {
            "title": "<nbsp>",
          },
        ],
        "title": "What is the correct HTML tag for inserting a line break?",
      },
      {
        "answers": [
          {
            "title": "url="http://mcqsets.com">MCQ Sets Quiz",
          },
          {
            "title": "<a name="http://mcqsets.com">MCQ Sets Quiz</a>",
          },
          {
            "isCorrect": true,
            "title": "<a href="http://mcqsets.com">MCQ Sets Quiz</a>",
          },
          {
            "title": "<http://mcqsets.com</a>",
          },
        ],
        "title": "What is the correct HTML for making a hyperlink?",
      },
      {
        "answers": [
          {
            "title": "None of above",
          },
          {
            "isCorrect": true,
            "title": "ASCII text",
          },
          {
            "title": "Special binary format",
          },
          {
            "title": "Machine language codes",
          },
        ],
        "title": "HTML documents are saved in",
      },
    ],
  },
  {
    "name": "Cars",
    "questions": [
      {
        "answers": [
          {
            "isCorrect": true,
            "title": "MINI",
          },
          {
            "title": "Morris Minor",
          },
          {
            "title": "Austin Metro",
          },
        ],
        "title": "What is the best-selling British car model of all time, known for its simple design and affordability?",
      },
      {
        "answers": [
          {
            "title": "Bentley",
          },
          {
            "isCorrect": true,
            "title": "Aston Martin",
          },
          {
            "title": "Lotus",
          },
        ],
        "title": "Which British car brand produced the iconic DB5, featured in James Bond films such as "Goldfinger"?",
      },
      {
        "answers": [
          {
            "title": "Stirling Moss",
          },
          {
            "isCorrect": true,
            "title": "Paddy Hopkirk",
          },
          {
            "title": "Jackie Stewart",
          },
        ],
        "title": "The Mini Cooper S is known for its rally successes in the 1960s. What is the name of the racing legend associated with driving the Mini?",
      },
      {
        "answers": [
          {
            "title": "Austin Allegro",
          },
          {
            "title": "Vauxhall Cavalier",
          },
          {
            "isCorrect": true,
            "title": "Ford Cortina Stewart",
          },
        ],
        "title": "Although the Ford Escort was incredibly popular, it wasn’t the best-selling family saloon of the 1970’s. Which of the following was?",
      },
      {
        "answers": [
          {
            "isCorrect": true,
            "title": "Spirit of Ecstasy",
          },
          {
            "title": "Flying Lady",
          },
          {
            "title": "Elegant Emissary",
          },
        ],
        "title": "The ornament on the bonnet of a Rolls Royce is called…",
      },
      {
        "answers": [
          {
            "isCorrect": true,
            "title": "Land Rover",
          },
          {
            "title": "Range Rover",
          },
          {
            "title": "Jaguar",
          },
        ],
        "title": "The Defender is a legendary off-road vehicle produced by which British automaker?",
      },
      {
        "answers": [
          {
            "title": "McLaren",
          },
          {
            "isCorrect": true,
            "title": "Lotus",
          },
          {
            "title": "Aston Martin",
          },
        ],
        "title": "Which British manufacturer makes the model ‘Elise’?",
      },
    ],
  },
]
`;
